File: src/main/scala/com/vodprofessionals/socialexplorer/vaadin/DashboardUIProvider.scala
package com.vodprofessionals.socialexplorer.vaadin

import com.vaadin.server.{UIClassSelectionEvent, UIProvider}
import com.vaadin.ui.UI

/**
 *
 */
class DashboardUIProvider extends UIProvider {
  override def getUIClass(event: UIClassSelectionEvent): Class[_ <: UI] = {
    // We only have one UI now, later might need a mobile UI or somethign
    classOf[DashboardUI]
  }
}

File: src/main/java/com/vodprofessionals/socialexplorer/vaadin/DashboardUIProvider.java
package com.vodprofessionals.socialexplorer.vaadin;

import com.vaadin.server.UIClassSelectionEvent;
import com.vaadin.server.UIProvider;
import com.vaadin.ui.UI;

public class DashboardUIProvider extends UIProvider {

    @Override
    public Class<? extends UI> getUIClass(UIClassSelectionEvent event) {
        return DashboardUI.class;
    }
}